webhook-service/lib/curl_validator.go

```diff
@@ -0,0 +1,44 @@
+package lib
+
+import (
+	"fmt"
+	"strings"
+)
+
+type CurlValidator struct {
+	denyListProvider IDenyListProvider
+	ipResolver       IIPResolver
+}
+
+type ICurlValidator interface {
+	Validate(request Request) error
+}
+
+func NewCurlValidator(denyListProvider IDenyListProvider, ipResolver IIPResolver) CurlValidator {
+	validator := CurlValidator{
+		denyListProvider: denyListProvider,
+		ipResolver:       ipResolver,
+	}
+	return validator
+}
+
+func (c CurlValidator) Validate(request Request) error {
+	if request.URL == "" {
+		return fmt.Errorf("Invalid curl URL: '%s'", request.URL)
```

Comment:
pls start the phrase with lowercase

Reply:
also, the URL will be always empty at this point, so I suggest to change the message to something like `request contains empty URL`